doc/development.md

```diff
@@ -89,4 +89,205 @@ merged, we need to test them on all supported distributions and
 architectures.
 
 When you push patches to GitHub, CI will run its tests according to the
-configuration in the `.github/workflows/ci.yml` file.
\ No newline at end of file
+configuration in the `.github/workflows/ci.yml` file.
+
+## Coding guidelines
+
+The coding conventions in Vdsm are generally aligned with the
+[PEP 8](https://peps.python.org/pep-0008/) standard guidelines.
+Code is linted with `pylint` based in the `/.pylintrc` file.
+You can lint the code before submitting a patch by doing:
+
+    make lint
+
+However, some code conventions are not explicitly enforced by the linter due
+to legacy code and inconsistent styles. But new code shall adhere to those
+rules, and old code, should be updated as possible when patched.
+
+In this section, we will cover some of the most important conventions and
+preferences for contributing to Vdsm.
```

Comment:
Before we deal with details, we need to give the general rules that apply to everything:

When modifying existing module, use the style of the current module. When adding a new module, use the style used by other modules in this package.